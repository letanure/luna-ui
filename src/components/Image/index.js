import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Image component.
 * E.g.:
 * ```html
 * <Image src="./theta.svg" alt="theta" />
 * ```
 */
class ImageComponent extends React.PureComponent {
  render() {
    const { src, alt, height, width, round, fit, shadow, className } = this.props;

    const Img = styled.img`
      width: ${width};
      height: ${height};
      ${fit ? 'object-fit: cover;' : ''}
      ${shadow ? 'box-shadow: 1px 1px 1px #838383;' : ''}
      ${round ? `border-radius: 100%; width: ${height}` : ''}
    `;

    return (
      <Img className={className} src={src} alt={alt}/>
    );
  }
}

ImageComponent.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  height: PropTypes.string,
  width: PropTypes.string,
  round: PropTypes.bool,
  /**
   * set object-fit as cover
   */
  fit: PropTypes.bool,
  shadow: PropTypes.bool,
  className: PropTypes.string
};

ImageComponent.defaultProps = {
  alt: 'image',
  height: 'auto',
  width: 'auto',
  round: false,
  fit: false,
  shadow: false,
  className: ''
};

export default ImageComponent;
