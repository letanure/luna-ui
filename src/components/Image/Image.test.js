import React from 'react';
import renderer from 'react-test-renderer';

import Image from './index';

describe('<Image />', () => {
  test('toMatchSnapshot: no attrs', () => {
    const component = <Image width="150px" height="150px" alt="Luna" src="./luna.jpg" />;
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: with attrs', () => {
    const component = <Image fit round shadow width="150px" height="150px" alt="Luna" src="./luna.jpg" />;

    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
