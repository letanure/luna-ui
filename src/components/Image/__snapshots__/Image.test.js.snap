// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> toMatchSnapshot: no attrs 1`] = `
<img
  alt="Luna"
  className="sc-bdVaJa kOpOcf"
  src="./luna.jpg"
/>
`;

exports[`<Image /> toMatchSnapshot: with attrs 1`] = `
<img
  alt="Luna"
  className="sc-bwzfXH MLNAr"
  src="./luna.jpg"
/>
`;
